### PR TITLE
Update code style.

### PR DESCRIPTION
**Pull request by drupol**
Update general code style:
* Remove unneeded imports.
* Add missing properties
* Change conditional structures and use early returns (or early continue)
* Use `::class` whenever it's possible.
* Remove unneeded variables.
* Replace `call_user_func_array()` with something faster.

### CODE REVIEW

**Comment by drupol** [comment]
@cweagans Could you review this ?

File: src/Resolvers/DependencyPatches.php [hunk 1]
```diff
@@ -27,18 +25,24 @@ public function resolve(PatchCollection $collection, PackageEvent $event)
...
         $operations = $event->getOperations();
         foreach ($operations as $operation) {
-            if ($operation->getJobType() == 'install' || $operation->getJobType() == 'update') {
-                // @TODO handle exception.
-                $package = $this->getPackageFromOperation($operation);
-                /** @var PackageInterface $extra */
-                $extra = $package->getExtra();
-                if (isset($extra['patches'])) {
-                    $patches = $this->findPatchesInJson($extra['patches']);
-                    foreach ($patches as $package => $patch_list) {
```
Comment (cweagans) [inline src/Resolvers/DependencyPatches.php]: While we didn't use the `$package` var here, I think this style provided some clarity around the data structure. Can you revert back to this notation?

Comment (cweagans) [reply src/Resolvers/DependencyPatches.php]: https://www.facebook.com/groups/millennialshow/permalink/2074484859516440/

Comment (cweagans) [reply src/Resolvers/DependencyPatches.php]: (just the `foreach` line - the rest of the style changes are fine)

File: src/Plugin/Patches.php [hunk 2]
```diff
@@ -76,6 +76,11 @@ class Patches implements PluginInterface, EventSubscriberInterface, Capable
      */
     protected $patchCollection;
 
+    /**
+     * @var array
+     */
+    private $installedPatches;
```
Comment (cweagans) [inline src/Plugin/Patches.php]: Was this mistakenly added?

Comment (drupol) [reply src/Plugin/Patches.php]: Nope, `$this->installedPatches` is used on line 97 of the same file :-)

File: src/Resolvers/DependencyPatches.php (hunk 1 above)
Comment (drupol) [reply src/Resolvers/DependencyPatches.php]: I updated it, changed the variable into something more meaningful without wasting memory with unused vars.

**Comment by drupol** [comment]
@cweagans Good for you ?

**Comment by cweagans** [comment]
Most of this is obsolete. See #447 for latest.

**Comment by drupol** [comment]
Thanks for being so responsive!

**Pull request closed without merging**